sklearn/utils/testing.py

```diff
@@ -882,3 +903,147 @@ def check_docstring_parameters(func, doc=None, ignore=None, class_name=None):
             if n1 != n2:
                 incorrect += [func_name + ' ' + n1 + ' != ' + n2]
     return incorrect
+
+
+def _check_matching_docstrings(doc_list, type_dict, type_name, object_name,
+                               include, exclude):
+    """
+    Checks if the docstring element, Parameter/Attribute/Return, having the
+    same name as a key in ``type_dict``, also has the same type definition and
+    description as that in ``type_dict``.
+
+    If a matching key is not found in ``type_dict``, the docstring element is
+    added in it with it's name as the key and value being a dictionary of it's
+    type_definition and description.
+
+    """
+    for name, type_definition, description in doc_list:
+        if exclude is not None and name in exclude:
+            pass
+        elif include is not True and name not in include:
+            pass
+        else:
+            # remove all whitespaces
+            type_definition = " ".join(type_definition.split())
+            description = [" ".join(s.split()) for s in description]
+            try:
+                description.remove('')
```

Comment:
This will only remove the first. You could use `list(filter(None, description))`